stitchee/concatenate.py

```diff
@@ -139,6 +140,104 @@ def validate_concat_method_and_dim(concat_method: str, concat_dim: str | None =
         )
 
 
+def find_variables_with_duplicate_dimensions(dataset: xr.DataSet):
+    # find variables with duplicate dimensions
+    # by comparing dimension's index with its first occurrence
+    duplicate_variables = [
+        varname
+        for varname in dataset.variables
+        if (
+            (dims := dataset[varname].dims)
+            and (any(idim != dims.index(dim) for idim, dim in enumerate(dims)))
+        )
+    ]
+
+    return duplicate_variables
+
+
+def rename_to_uniq_dimensions(dataarray):
+    # find the repetition numbers for all dimensions
+    iduplicates = [idim - dataarray.dims.index(dim) for idim, dim in enumerate(dataarray.dims)]
+    # rename repeated dimensions with a prefix to each repeated dimension
+    new_dims = [
+        dim if idup == 0 else f"{idup}___{dim}" for idup, dim in zip(iduplicates, dataarray.dims)
+    ]
+    dim_tuple = list(zip(dataarray.dims, new_dims))
+    # update coordinates with newly renamed dimensions
+    new_coords = {
+        new_dim: dataarray.coords[old_dim].rename({old_dim: new_dim})
+        for old_dim, new_dim in dim_tuple
+        if old_dim in dataarray.coords
+    }
+
+    # construct new DataArray with repeated dimensions
+    new_dataarray = xr.DataArray(
+        dataarray.to_numpy(),
+        dims=new_dims,
+        coords=new_coords,
+        name=dataarray.name,
+        attrs=dataarray.attrs,
+    )
+
+    return new_dataarray
+
+
+def concat_manually(dataarrays, concat_dim: str, concat_kwargs: dict):
+    # Build base kwargs
+    base_kwargs = {**DEFAULT_XARRAY_SETTINGS, **concat_kwargs}
+    # data_vars must be "all" when concatenating DataArrays because there is only one data_var, DataArray itself
+    base_kwargs.update({"data_vars": "all"})
+
+    # construct a list of dataarrays with repeated dimensions renamed
+    new_dataarrays = [rename_to_uniq_dimensions(da) for da in dataarrays]
+    # concatenate the manually constructed dataarrays
+    concatenated_dataarray = xr.concat(new_dataarrays, dim=concat_dim, **base_kwargs)
```

Comment:
Seems to me like only one comment line is necessary:
```suggestion
    # Rename duplicate dimensions, then concatenate
    new_dataarrays = [rename_to_uniq_dimensions(da) for da in dataarrays]
    concatenated_dataarray = xr.concat(new_dataarrays, dim=concat_dim, **base_kwargs)
```